wallet/src/Pos/Wallet/Web/Methods/Misc.hs

```diff
@@ -131,3 +143,48 @@ instance Buildable WalletStateSnapshot where
 
 dumpState :: MonadWalletWebMode m => m WalletStateSnapshot
 dumpState = WalletStateSnapshot <$> getWalletStorage
+
+----------------------------------------------------------------------------
+-- Print pending transactions info
+----------------------------------------------------------------------------
+
+data PendingTxsSummary = PendingTxsSummary
+    { ptiSlot    :: !SlotId
+    , ptiCond    :: !CPtxCondition
+    , ptiInputs  :: !(NonEmpty TxIn)
+    , ptiOutputs :: !(NonEmpty TxOut)
+    } deriving (Eq, Show, Generic)
+
+deriveJSON defaultOptions ''PendingTxsSummary
+
+instance Buildable PendingTxsSummary where
+    build PendingTxsSummary{..} =
+        bprint (  "  slotId: "%build%
+                "\n  status: "%build%
+                "\n  inputs: "%listJson%
+                "\n  outputs: "%listJson)
+            ptiSlot
+            ptiCond
+            ptiInputs
+            ptiOutputs
+
+instance HasTruncateLogPolicy PendingTxsSummary where
+    -- called rarely, and we are very interested in the output
+    truncateLogPolicy = identity
+
+gatherPendingTxsSummary :: MonadWalletWebMode m => m [PendingTxsSummary]
+gatherPendingTxsSummary =
+    map mkInfo .
+    getNewestFirst . toNewestFirst . sortPtxsChrono .
+    filter unconfirmedPtx <$>
+    getPendingTxs
+  where
+    unconfirmedPtx = not . isPtxInBlocks . _ptxCond
```

Comment:
Yes, this selects `PtxApplying` and `PtxWontApply`, but not `PtxInNewestBlocks` or `PtxPersisted`.